config/apply/ccsm/formatCards.js

```diff
@@ -0,0 +1,133 @@
+import { simpleResolver } from 'encender';
+
+export function formatCards(actions, otherResources, cards=[]) {
+  // For each action in the array
+  for (const action of actions) {
+    console.log('action: ', action);
+    // If action.resource.reference contains a CommunicationRequest
+    if (action.resource?.reference && action.resource?.reference?.includes('CommunicationRequest')) {
+      // Make an information card with that action (pass into extractInformation)
+      cards.push(extractInformation(action, otherResources));
+    // If action.selection behavior exists
+    } else if (action.selectionBehavior) {
+      // Make a suggestion card with the children actions as suggestions (pass children into extractSuggetsions)
+      // If child action is a Service Request, add a suggestion with an action with a resource
+      cards.push(extractSuggestions(action, otherResources));
+    // Else if sub-action exists, call extractCard on the sub-action
+    } else if (action.action) {
+      cards = formatCards(action.action, otherResources, cards);
+    }
+  }
+  return cards;
+}
+
+function extractInformation(action, otherResources) {
+  let resolver = simpleResolver([...otherResources], true);
+  let sources = action?.documentation ? getSources(action.documentation) : [{label:'no source listed'}];
+  let card = {
+    summary: action.title,
+    uuid: action.id, // Cards must have a uuid to render properly
+    indicator: getIndicator(action.priority),
+    source: sources[0],
+    links: sources.slice(1)?.map(s => ({...s,type:'absolute'})),
+    // Communication Request payload makes up the main contents of the card
+    detail: resolver(action.resource.reference)[0]?.payload[0].contentString ?? null
+  }; 
+  return card;
+}
+
+function extractSuggestions(action, otherResources) {
+  let resolver = simpleResolver([...otherResources], true);
+  let sources = action?.documentation ? getSources(action.documentation) : [{label:'no source listed'}];
+  let suggestions = [];
+  for (const subaction of action.action) {
+    if (subaction.resource?.reference?.includes('ServiceRequest')) {
+      let rsrc = resolver(subaction.resource.reference)[0] ?? {};
+      // Add necessary fields to Service Request resource
+      rsrc.status = 'draft';
+      if (rsrc?.code?.display) {
+        rsrc.code.text = rsrc.code.display;
+        delete rsrc.code.display;
+      }
+      if (!rsrc?.category) {
+        rsrc.category = [
+          {
+            coding: [
+              {
+                system: 'http://terminology.hl7.org/CodeSystem/medicationrequest-category',
+                code: 'inpatient',
+                display: 'Inpatient'
+              }
+            ]
+          }
+        ];
+      }
```

Comment:
Should we mandate that the category is inpatient, rather than only adding the inpatient category code if it doesn't exist?

```suggestion
    rsrc.category = [
      {
        coding: [
          {
            system: 'http://terminology.hl7.org/CodeSystem/medicationrequest-category',
            code: 'inpatient',
            display: 'Inpatient'
          }
        ]
      }
    ];
```